sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -220,7 +221,15 @@ private[hive] class HiveClientImpl(
     hiveConf
   }
 
-  private val userName = conf.getUser
+  private val userName = {
+    val doAs = sys.env.get("HADOOP_USER_NAME").orNull
+    val ugi = if (doAs != null && doAs.length() > 0) {
```

Comment:
This is wrong or, at the very least, backwards: the current UGI should be preferred.

What's the goal of this code? Why can't you just get the current UGI's name?

Reply:
Uh, at least that is not a regression, as conf.getUser calls Hive side and Hive does it in `Utils.getUGI`, at least from Hive 1.2.1.spark2.
So the code is copied and pasted to modify calling `ugi.getShortUserName()` instead of `ugi.getUserName()` - @hddong left a comment before to explain why the code had to be copied and pasted - https://github.com/apache/spark/pull/23952#discussion_r317426204

Reply:
Right, and there's a huge comment in the Hive method (at least in the branch I'm looking at) that explains why that is done. And if you read that comment you'll see that it does not apply here.

In a way, the HMS API is broken in that it lets the caller set the owner (instead of getting it from the auth info).

But we really should at least try to get the correct information through, and that comes from the UGI, not from env variables.

Reply:
Yeah actually I just read decompiled code of `Utils.getUGI()` and didn't indicate the intention - let other application be able to pass it. Yes I agree it's not needed in Spark side, and it would be weird `HADOOP_USER_NAME` is only used here and undocumented. Thanks for explaining!

Reply:
Thanks for you two, I missed to read comment in source too. Yes, current UGI is ok here,  SPARK-22846 Fix table owner is null when creating table through spark sql or thriftserver, but lead to      an issue that have occurred(owner is principal in kerberized cluster).